Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -486,6 +486,13 @@ void LocalFrameView::setFrameRect(const IntRect& newRect)
     if (m_frame->isMainFrame() && m_frame->page())
         m_frame->page()->pageOverlayController().didChangeViewSize();
 
+#if !PLATFORM(IOS_FAMILY)
+    if (auto* document = m_frame->document()) {
+        if (document->isImageDocument())
+            downcast<ImageDocument>(*document).windowSizeChanged();
```

Comment:
The traditional way to implement this type of thing would be a virtual function in `Document`. Could we use that here?

Reply:
Good idea. That would feel better. Less ad-hoc.